test/select.jl

```diff
@@ -702,7 +706,10 @@ end
                  [:x1, :x2] => (+) => :x1, 1:2 => ByRow(/) => :x3, :x1 => :x4, copycols=false)
     @test propertynames(df2) == [:x2, :x1, :x3, :x4, :r1, :r2]
     @test df.x2 === df2.x2
-    @test df.x1 === df2.x4
+    @test df.x1 == df2.x4
+    @test df.x1 == df2.x4
+    # a copy is made as we earlier used ":" selector, although later :x1 gets overwritten
```

Comment:
@nalimilan - here and below we have the following situation:
1. we process transforms sequentially
2. first a `:` selector is used so `:x1` column is marked as used
3. next we do `:x1 => :x4` so a copy is made although `copycols=false`

The issue is that in the mean time `[:x1, :x2] => (+) => :x1` operation is performed which overwrites the `:x1` column (that was marked as used). We do not remove a mark in such cases as it would be too complex IMO.